types/src/trusted_state.rs

```diff
@@ -182,7 +182,11 @@ impl TrustedState {
             // The ValidatorChangeProof is empty, stale, or only gets us into our
             // current epoch. We then try to verify that the latest ledger info
             // is this epoch.
-            self.verifier.verify(latest_li)?;
+            // TODO: store and verify the root hash if the version equals once new_trust_any_genesis_WARNING_UNSAFE
+            // removed
+            if latest_li.ledger_info().version() > self.latest_version {
+                self.verifier.verify(latest_li)?;
+            }
```

Comment:
Does this open up a vulnerability where a server can return a ledger info at client known version and skip ledger info signature checking?

Reply:
I think as long as the ledger info has same hash as the one client knows, it's safe regardless of the signatures

Reply:
Where do we check that today?

Reply:
There's a beautiful TODO I added, the blocker is the usage of new_trust_any_genesis_WARNING_UNSAFE, I can't initialize a hash for this function until we remove this function

Reply:
hmm, yeah the server could return random garbage in the `latest_li` but with `version == client_known_version` and the client would gladly accept it. the trusted_state is not compromised, but someone using the `latest_li` in `TrustedStateChange::Version` might read garbage. maybe we just have a `TrustedStateChange::NoChange` variant, or `Version` has an `Option<&LedgerInfo>`?